### PR TITLE
restructure SDK

### PR DESCRIPTION
**Pull request by aorfanos**

### CODE REVIEW

File: mailersend/base/base.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import json
+import os
+import requests
+
+API_BASE = "https://api.mailersend.com/v1"
+API_KEY = os.environ.get("MAILERSEND_API_KEY")
```
Comment (robgordon89) [inline mailersend/base/base.py:6]: not used?

File: mailersend/base/base.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    def __init__(
+        self,
+        api_base=None,
+        headers_default=None,
+        headers_auth=None,
+        mailersend_api_key=None,
+    ):
+
+        self.api_base = "https://api.mailersend.com/v1"
+        self.mailersend_api_key = os.environ.get("MAILERSEND_API_KEY")
```
Comment (robgordon89) [inline mailersend/base/base.py:19]: should use const above here

File: mailersend/base/base.py (hunk 1 above)
Comment (robgordon89) [reply mailersend/base/base.py:6]: is it also possible to offload this to the client, this gives the developer the best of both worlds, if they want to use an ENV they can pass it in 

see boto as a good example of how AWS do credentials

File: mailersend/base/base.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        self.headers_auth = f"Bearer {self.mailersend_api_key}"
+        self.headers_default = {
+            "Content-Type": "application/json",
+            "X-Requested-With": "XMLHttpRequest",
+            "User-Agent": "MailerSend-Client-python-v1",
+            "Authorization": f"{self.headers_auth}",
+        }
+
+        # super(NewAPIClient, self).__init__()
```
Comment (robgordon89) [inline mailersend/base/base.py:28]: is this required?

File: mailersend/emails/emails.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    def __init__(self):
+        baseObj = base.NewAPIClient()
+        super(NewEmail, self).__init__(
+            baseObj.api_base,
+            baseObj.headers_default,
+            baseObj.headers_auth,
+            baseObj.mailersend_api_key,
+        )
+
+    def setMailFrom(self, mail_from, message):
```
Comment (robgordon89) [inline mailersend/emails/emails.py:17]: from and recipients have email and name

File: mailersend/token/token.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        self.tokenScopes = tokenScopes
+
+        _data = {"name": tokenName, "scopes": tokenScopes}
+
+        request = requests.post(
+            f"{self.api_base}/token", headers=self.headers_default, json=_data
+        )
+        return request.text
+
+    def freezeToken(self, tokenId, pause=True):
```
Comment (robgordon89) [inline mailersend/token/token.py:26]: updateToken should be used here

File: README.md [hunk 6]
```diff
@@ -5,219 +5,722 @@ MailerSend Python SDK
 [![MIT licensed](https://img.shields.io/badge/license-MIT-blue.svg)](./LICENSE.md)
 
 # Table of Contents
+
 * [Installation](#installation)
 * [Usage](#usage)
-* [Testing](#testing)
+  * [Sending and email](#sending_an_email)
+  * [Sending and email with CC and BCC](#cc_and_bcc)
```
Comment (robgordon89) [inline README.md:12]: please update check all headings should match CURL header (H3) in doc

File: README.md [hunk 7]
```diff
@@ -5,219 +5,722 @@ MailerSend Python SDK
...
+
+```bash
+python -m pip install mailersend
 ```
 
 <a name="usage"></a>
+
 # Usage
 
-### Requires the `MAILERSEND_API_KEY` environment variable
```
Comment (robgordon89) [inline README.md:24]: MAILERSEND_API_KEY removed?

File: README.md [hunk 8]
```diff
@@ -5,219 +5,722 @@ MailerSend Python SDK
...
+        "name": "Client1",
+        "email": "yetanother@client.com"
     },
     {
-        "var": "company",
-        "value": "Dunder Mifflin",
-    },]
-}]
+        "name": "Client2",
+        "email": "justanother@client.com"
```
Comment (robgordon89) [inline README.md:134]: could we try and keep emails the same throughout our SDK's 

https://github.com/mailersend/mailersend-go

cc@client.com
bcc@client.com

etc

File: README.md [hunk 9]
```diff
@@ -5,219 +5,722 @@ MailerSend Python SDK
...
+    'your@client.com',
+    'another@client.com
+]
+
+variables = [
+    {
+        "email": "your@client.com",
+        "substitutions": [
+            {
+                "var": "company",
```
Comment (robgordon89) [inline README.md:177]: use foo / bar for variables

File: README.md [hunk 10]
```diff
@@ -5,219 +5,722 @@ MailerSend Python SDK
...
+mail_body = {}
+
+recipients = [
+    'your@client.com',
+    'another@client.com
+]
+
+personalization = [
+    {
+        "email": "test@mailersend.com",
```
Comment (robgordon89) [inline README.md:217]: this email should match a recipient (or it wont do anything)

**Review by robgordon89 — CHANGES_REQUESTED** [review]
A few inline comments on this I also wanted to ask the UX around this sdk seems quite different than the rest of our SDK's for instances why 

```python
import mailersend.webhooks.webhooks as webhooks
```

why not 

```python
import mailersend.webhooks as webhooks 
```

I guess this is down to the package structure but it does look strange with these imports 🤔

File: mailersend/base/base.py (hunk 3 above)
Comment (aorfanos) [reply mailersend/base/base.py]: comment removed 👍🏼

File: mailersend/emails/emails.py (hunk 4 above)
Comment (aorfanos) [reply mailersend/emails/emails.py]: fields now accept dict and list of dicts respectively

File: mailersend/token/token.py (hunk 5 above)
Comment (aorfanos) [reply mailersend/token/token.py]: function name changed

File: README.md (hunk 6 above)
Comment (aorfanos) [reply README.md]: readme updated according to Golang SDK

File: README.md (hunk 7 above)
Comment (aorfanos) [reply README.md:24]: added in doc

**Review by robgordon89 — APPROVED** [review]

**Comment by scripterkaran** [comment]
Can we merge this? Need support for python 3.7 and I see that this PR adds it.

**Pull request merged**